### ISSUE TITLE
Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project cloud-utils

### ISSUE THREAD

**Issue by inscil** [apache/cloudstack#5016]
##### ISSUE TYPE

 * Other

##### COMPONENT NAME

~~~
BUILD FAILURE
~~~

##### CLOUDSTACK VERSION

~~~
4.14.1.0
4.15.0.0
master on 20210511
~~~

##### OS / ENVIRONMENT

Debian 10

maven 3.6.0-1
libmaven-parent-java 31-2
libmaven-resolver-java 1.3.1-1
libmaven-shared-utils-java 3.3.0-1
libmaven3-core-java 3.6.0-1

openjdk-11-jdk:amd64 11.0.9+1-1 or 11.0.11+9-1
openjdk-11-jdk-headless:amd64 11.0.9+1-1 or 11.0.11+9-1
openjdk-11-jre:amd64 11.0.9+10-1 or 11.0.11+9-1
openjdk-11-jre-headless:amd64 11.0.9+1-1 or 11.0.11+9-1

##### SUMMARY

There are test failures during build.

Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project cloud-utils: There are test failures.

##### STEPS TO REPRODUCE

Follow the latest documentation (4.15.0) on how to build CloudStack from source on a Debian 10 system.

##### ACTUAL RESULTS

[debug_build.log.gz](https://github.com/apache/cloudstack/files/6459715/debug_build.log.gz)
[surefire-reports.tar.gz](https://github.com/apache/cloudstack/files/6459716/surefire-reports.tar.gz)

##### OTHER INFO
The offending bit seems to be in TEST-com.cloud.utils.rest.BasicRestClientTest.xml:
Unexpected exception, expected<com.cloud.utils.rest.CloudstackRESTException> but was<java.lang.AssertionError>

**Comment by rhtyd** [comment]
@inscil can you try building on Ubuntu? Or building with a `-DskipTests=true`. I think the failure maybe related to env, as Travis and our packing jobs are passing on 4.15 and master branch.

**Comment by inscil** [comment]
Building on Debian with _-DskipTests=true_ yields the same results - as does trying to build in Ubuntu 20.04 (same versions for .
openjdk and maven). I also tried using openjdk-8, as the documentation [here](https://docs.cloudstack.apache.org/en/4.15.0.0/installguide/building_from_source.html#extracting-source) suggests. This didn't go down well at all.

Admittedly, I am somewhat confused about which versions of python and java and mysql-connector are needed. In Debian I tried with python 2.7 and 3.7 (and of course the respective connector). Same result. Could this still be the problem?

**Comment by rhtyd** [comment]
@inscil pl refer to https://github.com/shapeblue/hackerbook/blob/master/2-dev.md#setting-up-development-environment written/tested against JDK11 and Ubuntu 20.04. If you've further question, pl join the dev@ ML and ask questions there: http://cloudstack.apache.org/mailing-lists.html

**Comment by inscil** [comment]
@rhtyd Thank you for the link. With it I was able to figure out a way to build the src on Debian 10.

If interesting for anyone, I did the following on a newly installed system:
~~~
apt install openjdk-11-jdk maven mariadb-server mariadb-client bzip2 nfs-common uuid-runtime python2 python2-dev python-setuptools ipmitool genisoimage nfs-kernel-server quota git devscripts dh-systemd nodejs npm

curl -sL https://deb.nodesource.com/setup_12.x | bash -
apt install nodejs

wget http://dev.mysql.com/get/mysql-apt-config_0.8.17-1_all.deb
dpkg -i mysql-apt-config_0.8.17-1_all.deb
apt update
apt install mysql-connector-python-py3

Put mysql config from docs into /etc/mysql/mariadb.conf.d/50-server.cnf
~~~

To build the src I did:
~~~
git clone https://github.com/shapeblue/cloudstack-nonoss
git clone --branch 4.15 https://github.com/apache/cloudstack.git

In cloudstack-nonoss dir ==> bash -x install-non-oss.sh

In cloudstack dir:
change python3-mysql.connector to mysql-connector-python-py3 in file "debian/control"
mvn -P deps (I believe, this step is unnecessary but did not test it)
bash -x packaging/build-deb.sh
~~~

The get the management running on a different machine, in addition to the steps described in the documentation I had to install "genisoimage" and link its binary to "mkisofs".
